mart/attack/callbacks/progress_bar.py

```diff
@@ -15,10 +15,11 @@
 class ProgressBar(Callback):
     """Display progress bar of attack iterations with the gain value."""
 
-    def on_run_start(self, adversary, input, target, model, **kwargs):
+    def on_run_start(self, *, adversary, **kwargs):
         self.pbar = tqdm.tqdm(total=adversary.max_iters, leave=False, desc="Attack", unit="iter")
 
-    def on_examine_end(self, adversary, input, target, model, **kwargs):
+    def on_examine_end(self, *, adversary, **kwargs):
+        input = kwargs["input"]
```

Comment:
Just change this to:
```python
def on_examine_end(self, *, input, adversary, **kwargs):
```
You require input so just make it an parameter of the function!

Reply:
Fixed.